### PR TITLE
[Refactor][envtest] Centralize all helpers in envtest for better DX

### PR DESCRIPTION
**Pull request by rueian**
## Why are these changes needed?

We often time want to run only a subset of tests while developing. Instead of using complicated Ginkgo filters or something not obvious modifications like `FDescribe`, we usually want to simply delete unwanted test files temporarily. But these test helpers scattered around the Ginko suite block us from doing this because we may delete helpers that are used in the remaining tests.

This PR moves these helpers to a centralized file so that we can quickly omit unwanted tests with the method mentioned above without deleting needed helpers.

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: ray-operator/controllers/ray/suite_helpers_test.go [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+import (
+	"context"
+	"log"
+	"reflect"
+	"time"
+
+	"github.com/onsi/gomega"
+	v12 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
+	"github.com/ray-project/kuberay/ray-operator/controllers/ray/utils"
+	v1 "k8s.io/api/core/v1"
```
Comment (kevin85421) [inline ray-operator/controllers/ray/suite_helpers_test.go:12]: corev1?

File: ray-operator/controllers/ray/suite_helpers_test.go [hunk 2]
```diff
@@ -0,0 +1,232 @@
+package ray
+
+import (
+	"context"
+	"log"
+	"reflect"
+	"time"
+
+	"github.com/onsi/gomega"
+	v12 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
```
Comment (kevin85421) [inline ray-operator/controllers/ray/suite_helpers_test.go:10]: rayv1?

File: ray-operator/controllers/ray/suite_helpers_test.go (hunk 1 above)
Comment (rueian) [reply ray-operator/controllers/ray/suite_helpers_test.go]: 👍

File: ray-operator/controllers/ray/suite_helpers_test.go (hunk 2 above)
Comment (rueian) [reply ray-operator/controllers/ray/suite_helpers_test.go]: 👍

**Review by kevin85421 — APPROVED** [review]

**Pull request merged**